src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopeDistribution.h

```diff
@@ -0,0 +1,312 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Clemens Groepl, Andreas Bertsch, Chris Bielow $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_COARSEID_H
+#define OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_COARSEID_H
+
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternGenerator.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopeDistribution.h>
+
+
+namespace OpenMS
+{
+
+  /**
+    *         @ingroup Chemistry
+    *         @brief Isotope pattern generator for coarse isotope distributions.
+    *         
+    *         This algorithm generates theoretical pattern distributions for empirical
+    *         formulas with resolution of 1Da.
+    *         It assumes that every isotope has atomic mass that is rounded
+    *         to the closest integer in Daltons.
+    *         For example for (13)Carbon it assumes that the mass of the isotope is 13Da
+    *         instead of 13.0033548378
+    *
+    *         The most important value which should be set is the max isotope value.
+    *         It limits the number of calculated isotopes.
+    *         If e.g., set to 3, only the first three isotopes, Monoisotopic mass, +1 and +2 are
+    *         calculated.
+    *         By default all possible isotopes are calculated, which leads to a large
+    *         number of values, if the mass value is large!
+    *
+    *         See also method run()
+    *
+    *         Distributions can be added to one another or multiplied with an integer.
+    **/
+
+  class OPENMS_DLLAPI CoarseIsotopeDistribution 
+    : public IsotopePatternGenerator
```

Comment:
I dont think the naming makes sense, the `CoarseIsotopeDistribution` is not a distribution and has little to do with `IsotopeDistribution`, so why not rename it `CoarseIsotopePatternGenerator` which also makes a lot of sense given the inhertiance and the "is-a" relationship